### PR TITLE
feat: parse real-world .bat/.cmd fixtures

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

Adds 9 real-world fixtures (SysToolsLib batch scripts, Apache Maven `mvn.cmd`) to `examples/` and extends the grammar so 15 of 18 fixtures now parse with **zero ERROR nodes**. Driven by failing corpus tests first.

| Fixture | Before | After |
|---|---|---|
| `mvn.cmd` | 43 | **0** |
| `AutoRun.cmd` | 10 | **0** |
| `CheckEOL.bat` | 12 | **0** |
| `TimeX.bat` | 3 | **0** |
| `2note2.bat` | 19 | **0** |
| `Library.bat` | 139 | 29 |
| `paths.bat` | 58 | 41 |
| `zap.bat` | 3 | 3 |

Remaining errors live in `Library.bat` (dense DosTips macros with `%\n%` pseudo-newlines and caret-quoted SET with `%%~var` inside the name) and `paths.bat` (a true batch+PowerShell polyglot with PS code after the `#>` boundary). `zap.bat`'s three errors are an actual malformed line in upstream (`# Delete one directory.` — missing the `:` prefix).

## Grammar additions

- **Caret `^` line continuation** via `extras` — unblocks `mvn.cmd`'s 14-line java launcher and `set LF=^` LF-variable idiom
- **Quoted string as command / call target** — `"%JAVACMD%" -version`, `call "%PROGRAMDATA%\rc.cmd" %*`
- **`<&` fd redirect** + variable-handle redirects `>&%FD%` / `<&%FD%`
- **Redirect fd range widened to 0-9** and op token merged with the fd prefix; this also unblocks bare digits 0/1/2 (`echo 1`, `%RETURN% 0` no longer lose the digit to a phantom redirect prefix)
- **Delayed expansion with subscripts & nested `%%`**: `!ARR[%%i]!`, `!T%%n.YEAR!`
- **For-variable modifiers**: `%%~zS`, `%%~dpnxF`, `%%~$PATH:F`, throwaway `%%.`
- **Command name may start with a digit** (`1clip.exe`)
- **Variable name with `+` / `-` prefix** (`%+INDENT%`)
- **DosTips idioms**: `(call,)`, `(call;)`, bare `(call)`
- **`@(...)` parenthesized block prefix**
- **SET display-only** (no `=`), **SET /P** with empty var and quoted var name
- **SET value may carry a trailing redirection** (`set "X=Y" 2>NUL`)
- **SET unquoted name may mix varref + literal** (`set TITLE.%%a=%%c`)
- **`if` with multiple adjacent strings** (`"x"==""""`)
- **Literal `!` / `%` as single-char argument_value** (for `%X%!%~1!`)
- **`echo_off` participates in command_sep** (`@echo off & setlocal ...`)
- **Polyglot header tolerance**: `<# ...` header lines and lines ending in `'VBS` parse as comments so SysToolsLib-style dual-language scripts don't fight the batch section

## Tests & CI

- 15 new corpus files, 42 new cases (235 total, 100% passing)
- `.github/workflows/test.yml` asserts zero ERROR/MISSING on 8 real-world fixtures
- README Features section updated

## Test plan

- [x] `npx tree-sitter test` — 235/235 pass
- [x] `npm run lint` — clean
- [x] `npx tree-sitter parse examples/*.{bat,cmd} --quiet --stat` — parses 18 files including the three known-partial fixtures
- [x] `for f in examples/{search.bat,dope.bat,2note2.bat,AddPaths.bat,AutoRun.cmd,CheckEOL.bat,TimeX.bat,mvn.cmd}; do npx tree-sitter parse "$f" | grep -q ERROR && echo FAIL "$f" || echo OK "$f"; done` — all OK

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Expanded grammar support for polyglot batch/PowerShell headers and VBScript markers
  * Enhanced variable reference handling with subscript support and additional character sets
  * Added line continuation support via caret (`^`) 
  * Improved file descriptor redirection using variable references
  * Added 8 new example scripts demonstrating advanced batch utilities

* **Documentation**
  * Updated feature documentation reflecting broader syntax coverage

* **Tests**
  * Added 14 new test fixtures validating edge cases and enhanced features

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -45,6 +45,12 @@ export default grammar({
       seq(optional('@'), /[rR][eE][mM]/, optional(seq(/[ \t]/, /[^\r\n]*/))),
       seq(':', /[^$a-zA-Z_\r\n]/, /[^\r\n]*/),
       seq('%#', /[^\r\n]*/, '#%'),
+      // Polyglot batch/PowerShell header line: `<# ...` is a PS block-comment open
+      // that batch interprets as a (failing) redirect; we treat it as a comment.
+      seq('<#', /[^\r\n]*/),
+      // Batch+VBScript polyglot: lines ending with the `'VBS` marker are extracted
+      // by the script itself as VBScript; the batch interpreter never runs them.
+      seq(/[^'\r\n][^\r\n]*[ \t]'VBS/),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 9de0b83. Thanks @gemini-code-assist!

File: grammar.js [hunk 2]
```diff
@@ -82,7 +88,10 @@ export default grammar({
     prompt_assignment: ($) => seq(
       optional(/[ \t]+/), alias(ci('/p'), $.set_option),
       optional(/[ \t]+/),
-      alias(/[@a-zA-Z_][@a-zA-Z0-9_()\[\]]*/, $.variable_name), '=', optional($.assignment_value),
+      choice(
+        seq(optional(alias(/[@a-zA-Z_][@a-zA-Z0-9_()\[\]]*/, $.variable_name)), '=', optional($.assignment_value)),
+        seq('"', optional(alias(/[@a-zA-Z_][@a-zA-Z0-9_()\[\]]*/, $.variable_name)), '=', optional($.quoted_assignment_value), '"', optional($.argument_list)),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 3c4fb26. Thanks @gemini-code-assist!

File: examples/zap.bat [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+)
+:# If we were asked for a recursive action, do it recursively.
+if %RECUR%==1 (
+  call :zapRecur "%~1"
+  goto next_path
+)
+:# Else just delete files in the target (default: current) directory
+call :zapFiles "%~1"
+goto next_path
+
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply examples/zap.bat:514]: @coderabbitai files at `examples/` folder are test fixtures - don't review them

File: grammar.js [hunk 4]
```diff
@@ -188,7 +204,7 @@ export default grammar({
       choice($.parenthesized, $._body_stmt),
     )),
     for_options: () => token(prec(10, choice(ci('/d'), seq(ci('/r'), optional(seq(/[ \t]+/, /(%[^\s%]|[^\s%])+%?/))), ci('/l'), seq(ci('/f'), optional(seq(/[ \t]+/, '"', /[^"]*/, '"')))))),
-    for_variable: () => token(seq('%%', optional('~'), /[a-zA-Z]/)),
+    for_variable: () => token(seq('%%', optional(seq('~', /[a-zA-Z$:_]*/)), /[a-zA-Z.]/)),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 36420c2. Thanks @gemini-code-assist!

File: grammar.js [hunk 5]
```diff
@@ -21,12 +21,12 @@ const varRefChoice = () => choice(
...
   $.cond_exec, $.pipe_stmt, $.redirect_stmt, $.call_stmt, $.cmd, $.parenthesized,
   $.variable_assignment, $.goto_stmt, $.exit_stmt, $.setlocal_stmt, $.endlocal_stmt,
-  $.if_stmt, $.for_stmt, $.macro_invocation,
+  $.if_stmt, $.for_stmt, $.macro_invocation, $.echo_off,
 ];
 
 export default grammar({
   name: 'batch',
-  extras: () => [/[ \t]/],
+  extras: () => [/[ \t]/, token(prec(20, /\^\r?\n[ \t]*/))],
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in d02683c. Thanks @coderabbitai!

File: grammar.js [hunk 6]
```diff
@@ -2,15 +2,15 @@ const ci = (word) => new RegExp(word.split('').map((c) => /[a-zA-Z]/.test(c) ? `
...
+  seq('%', /[$@+\-a-zA-Z_][$@a-zA-Z0-9_.#()\[\]]*/, '%'),
   seq('%~', /[a-zA-Z]*/, /[0-9]/),
   seq('%', /[0-9]/),
   '%*',
-  seq('%%%%%%', optional('~'), /[a-zA-Z0-9]/),
-  seq('%%%%', optional('~'), /[a-zA-Z0-9]/),
-  seq('%%', optional('~'), /[a-zA-Z]/),
+  seq('%%%%%%', optional(seq('~', /[a-zA-Z$:_]*/)), /[a-zA-Z0-9]/),
+  seq('%%%%', optional(seq('~', /[a-zA-Z$:_]*/)), /[a-zA-Z0-9]/),
+  seq('%%', optional(seq('~', /[a-zA-Z$:_]*/)), /[a-zA-Z]/),
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 2da45cd. Thanks @gemini-code-assist!

File: grammar.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply grammar.js]: ✅ Addressed in 0ffd093. Thanks @coderabbitai!
